### PR TITLE
Introduce task for cleaning up accepted API changes

### PR DESCRIPTION
**Pull request by bmuschko**
### Context

This task is called when initiating the branching process. For more information see https://github.com/gradle/gradle-private/issues/1067.

### CODE REVIEW

File: buildSrc/src/test/groovy/org/gradle/binarycompatibility/AcceptedApiChangesJsonFileManagerTest.groovy [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        jsonFileManager.emptyAcceptedApiChanges(jsonFile)
+
+        then:
+        jsonFile.text == nonExistentAcceptedApiChanges()
+    }
+
+    static String existingAcceptedApiChanges() {
+        """
+            {
+                "baseVersion": "4.1-20170714213039+0000",
```
Comment (wolfs) [inline buildSrc/src/test/groovy/org/gradle/binarycompatibility/AcceptedApiChangesJsonFileManagerTest.groovy]: `baseVersion` isn't part of the file any more. How did you come up with the example?

File: buildSrc/src/main/groovy/org/gradle/binarycompatibility/AcceptedApiChangesJsonFileManager.groovy [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+package org.gradle.binarycompatibility
+
+import groovy.json.JsonBuilder
+import groovy.json.JsonSlurper
+
+class AcceptedApiChangesJsonFileManager {
+
+    void emptyAcceptedApiChanges(File jsonFile) {
+        def slurped = new JsonSlurper().parse(jsonFile)
```
[the remark this answers is not in the document]
Comment (bmuschko) [reply buildSrc/src/main/groovy/org/gradle/binarycompatibility/AcceptedApiChangesJsonFileManager.groovy]: I wasn't quite sure about the schema of the file. I was assuming that there are other key-value pairs in the file that needed to be preserved. My solution might actually be more future-proof and it doesn't wipe out everything else.

File: buildSrc/src/test/groovy/org/gradle/binarycompatibility/AcceptedApiChangesJsonFileManagerTest.groovy (hunk 1 above)
Comment (bmuschko) [reply buildSrc/src/test/groovy/org/gradle/binarycompatibility/AcceptedApiChangesJsonFileManagerTest.groovy]: Got this from the linked commit in the release document under ["Branching for the Release"](https://docs.google.com/document/d/1h_Me4H4PfCRNUg5KLHht9zdSMNmMeOmQycvSyeBtyJ8/edit#heading=h.3znysh7) (point 4): https://github.com/gradle/gradle/blob/d4133a7e85046ed2c6f677739043487f38a83898/subprojects/distributions/src/changes/accepted-public-api-changes.json#L4-L39. I will remove the `baseVersion` if that doesn't exist anymore.

**Comment by bmuschko** [comment]
> Nice test coverage!

Thanks!

**Pull request merged**